components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,958 @@
+        } catch (IllegalArgumentException | UnsupportedEncodingException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.error("Authorize URL creation failed with the additional query parameters issue.");
+            }
+            throw new UAEPassAuthnFailedException("Authentication process failed. Unable to set " +
+                    "additional query parameters to the authorize request.", e);
+        }
+
+        return finalAuthzUrl;
+    }
+
+    /**
+     * This method is used to retrieve user claims as key value pairs to the Java Map object from user info endpoint.
+     *
+     * @param oAuthResponse         The response from OAuthClient to authenticator by the UAEPass.
+     *                              (Use to teh get access token)
+     * @param context               The Authentication context received by authenticator.
+     * @return Map<String, Object>  Map object of key value pairs of the logged user.
+     */
+    public Map<String, Object> getUserInfoUserAttributes(OAuthClientResponse oAuthResponse,
+                                                         AuthenticationContext context)
+            throws UAEPassUserInfoFailedException {
+
+        String accessToken = oAuthResponse.getParam(UAEPassAuthenticatorConstants.UAE.ACCESS_TOKEN);
+        String userInfoJsonPayload;
+        Map<String, Object> userInfoJwtAttributes = null;
+
+        try {
+            userInfoJsonPayload = sendUserInfoRequest(context, accessToken);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Successfully returns the userinfo JSON payload");
+            }
+            Set<Map.Entry<String, Object>> jwtAttributeSet;
+            jwtAttributeSet = JSONObjectUtils.parse(userInfoJsonPayload).entrySet();
+            userInfoJwtAttributes = buildJSON(jwtAttributeSet);
+
+        } catch (UAEPassUserInfoFailedException e) {
+            throw new UAEPassUserInfoFailedException("Unable to retrieve claims from user info.", e);
+        } catch (ParseException e) {
+            LOG.error("Error occurred while parsing user info payload by UAEPass.");
+            throw new UAEPassUserInfoFailedException("Error occurred while parsing user info payload by UAEPass.", e);
+        }
+
+        return userInfoJwtAttributes;
+    }
+
+    /**
+     * This method is used to create userinfo request with the access token.
+     *
+     * @param context                          The Authentication context received by authenticator.
+     * @param accessToken                      The access token obtained from the processAuthenticationResponse.
+     * @return String                          The response which returns from the user info API call.
+     * @throws UAEPassUserInfoFailedException  Throws an exception, if not obtains the user claims from the user info.
+     */
+    public String sendUserInfoRequest(AuthenticationContext context, String accessToken)
```

Comment:
As we using this method only in the authenticator class, shall we make this private

Reply:
Fixed with : [c9f22571bc5](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/c9f22571bc5ffd21428c8b3a022c61e2045de40f)